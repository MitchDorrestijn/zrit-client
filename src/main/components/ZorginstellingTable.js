/**
 * React related imports
 */
import React from 'react';
import {Button} from 'reactstrap'
import {CSVLink} from 'react-csv';
import axios from 'axios';
import config from '../config';
import {
  BootstrapTable,
  TableHeaderColumn,
  SearchField
} from 'react-bootstrap-table';


/**
 * Style related imports
 */
import '../../../node_modules/react-bootstrap-table/dist/react-bootstrap-table-all.min.css';
import '../css/table.css';

/**
 * This class takes care of rendering the zorginstellingen table, from here other actions can be taken
 */
export default class ZorginstellingTable extends React.Component {

  componentDidMount(){
    axios.get(`${config.url}/zorginstelling/zorginstellingen`).then((res) => {
      this.setState({data: res.data});
    }).catch((err) => {
      console.log(err);
    });
  }

  /**
   * Setups all the data.
   * @constructor
   * @param {props} props - The given properties by the superclass
   */
  constructor(props) {
    super(props);
    this.columns = [
      {
        name: 'name',
        display: 'Naam'
      }
    ];
    this.state = {
      sortColumnName: undefined,
      sortOrder: undefined,
      disableButtons: true,
      data: []
    };
  };

  /**
   * Redirect to a page where a the selected zorginstelling can be updated.
   */
  updateItem = () => {
    return this.props.history.push(`${this.props.routes.updateZorginstelling}/${this.refs.table.state.selectedRowKeys}`);
  };

  /**
   * Redirect to a page where a new zorginstelling can be added.
   */
  createItem = () => {
    return this.props.history.push(this.props.routes.createZorginstelling);
  };

  /**
   * Enables the buttons for allowing the zorginstelling to update
   */
  onSelect = () => {
    this.setState({disableButtons: false});
  }

  /**
   * Renders the columns sorted.
   * @param {string} sortColumnName - Name of the column that is going to be sorted
   * @param {string} sortOrder - Can be either desc or asc
   */
  renderSortedColumn = (sortColumnName, sortOrder) => {
    this.setState({sortColumnName: sortColumnName, sortOrder: sortOrder});
  };

  /**
   * Renders the search field above the table
   */
  renderSearchField = () => {
    return (<SearchField className='searchfield' placeholder='Type om te zoeken'/>);
  };

  /**
   * Renders the buttons above the table
   */
  renderButtons = () => {
    return (<div>
      <Button onClick={this.createItem} color="primary" className='crud-btn'>Toevoegen</Button>
      <Button disabled={this.state.disableButtons}
        onClick={this.updateItem} color="primary" className='crud-btn'>Bewerken</Button>
    </div>);
  };

  /**
   * Renders the view for the user
   */
  render() {
    const tableOptions = {
      sortColumnName: this.state.sortName,
      sortOrder: this.state.sortOrder,
      renderSortedColumn: this.renderSortedColumn,
      noDataText: 'Geen resultaten gevonden',
      searchField: this.renderSearchField,
      btnGroup: this.renderButtons
    };

    return (
      <div>
        <BootstrapTable search={true} data={this.state.data} options={tableOptions} selectRow={{
            mode: 'radio',
            onSelect: this.onSelect
          }} ref='table'>

          <TableHeaderColumn hidden={true} dataField="id" isKey={true}>
            ID
          </TableHeaderColumn>

          <TableHeaderColumn dataField="name" dataSort={true}>
            Naam &#x2195;
          </TableHeaderColumn>
        </BootstrapTable>

        <CSVLink data={this.state.data} filename={"zorginstelling_overview"} className="btn btn-primary crud-btn" target="">
          Export als CSV
        </CSVLink>
      </div>
    );
  };
}
