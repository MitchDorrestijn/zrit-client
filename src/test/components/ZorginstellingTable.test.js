import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import ZorginstellingTable from '../../main/components/ZorginstellingTable';

const wrapper = shallow(<ZorginstellingTable history="" routes="" />);

const push = jest.fn();
wrapper.setProps({ history: { push } });

test('CRUDTable component should match snapshot', () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('Should render two TableHeaderColumn components', () => {
  expect(wrapper.find('TableHeaderColumn').length).toBe(2);
});

test('Should render one BootstrapTable component', () => {
  expect(wrapper.find('BootstrapTable').length).toBe(1);
});

test('Should render one CSVLink component', () => {
  expect(wrapper.find('CSVLink').length).toBe(1);
});
