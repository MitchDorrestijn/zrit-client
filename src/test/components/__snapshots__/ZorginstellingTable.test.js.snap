// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CRUDTable component should match snapshot 1`] = `
<div>
  <BootstrapTable
    autoCollapse={
      Object {
        "filter": false,
        "search": false,
        "sort": false,
      }
    }
    bodyContainerClass={null}
    bordered={true}
    cellEdit={
      Object {
        "afterSaveCell": undefined,
        "beforeSaveCell": undefined,
        "blurToEscape": false,
        "blurToSave": false,
        "mode": "none",
        "nonEditableRows": undefined,
      }
    }
    columnFilter={false}
    condensed={false}
    containerClass={null}
    csvFileName="spreadsheet.csv"
    data={
      Array [
        Object {
          "bestemming": "Herenlaan 23, 3701 AR, Zeist",
          "chauffeur_toegewezen": "Nee",
          "datum": "22-02-2017",
          "id": 1,
          "naam_chauffeur": "-",
          "naam_client": "Gerda Willems",
          "ophaal_locatie": "2e Hogeweg 46, 3709 AZ, Zeist",
        },
        Object {
          "bestemming": "Singelpark 7, 3984 NC, Odijk",
          "chauffeur_toegewezen": "Nee",
          "datum": "21-02-2017",
          "id": 2,
          "naam_chauffeur": "-",
          "naam_client": "Hans de Wit",
          "ophaal_locatie": "Wegedoorn 23, 3984 AZ, Odijk",
        },
        Object {
          "bestemming": "Herenlaan 23, 3701 AR, Zeist",
          "chauffeur_toegewezen": "Ja",
          "datum": "20-02-2017",
          "id": 3,
          "naam_chauffeur": "Taxi bedrijf",
          "naam_client": "Gerda Willems",
          "ophaal_locatie": "2e Hogeweg 46, 3709 AZ, Zeist",
        },
      ]
    }
    deleteRow={false}
    expandColumnOptions={
      Object {
        "expandColumnBeforeSelectColumn": true,
        "expandColumnComponent": undefined,
        "expandColumnVisible": false,
        "expandedColumnHeaderComponent": undefined,
      }
    }
    exportCSV={false}
    fetchInfo={
      Object {
        "dataTotalSize": 0,
      }
    }
    footer={false}
    headerContainerClass={null}
    height="100%"
    hover={false}
    ignoreSinglePage={false}
    insertRow={false}
    keyBoardNav={false}
    multiColumnSearch={false}
    multiColumnSort={1}
    options={
      Object {
        "btnGroup": [Function],
        "noDataText": "Geen resultaten gevonden",
        "renderSortedColumn": [Function],
        "searchField": [Function],
        "sortColumnName": undefined,
        "sortOrder": undefined,
      }
    }
    pagination={false}
    printable={false}
    renderAlert={true}
    replace={false}
    search={true}
    selectRow={
      Object {
        "mode": "radio",
        "onSelect": [Function],
      }
    }
    striped={false}
    tableBodyClass={null}
    tableContainerClass={null}
    tableFooterClass={null}
    tableHeaderClass={null}
    trClassName=""
    version="3"
    withoutTabIndex={false}
  >
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="id"
      dataSort={false}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      export={false}
      filterFormatted={false}
      headerTitle={true}
      hidden={true}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={true}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width={null}
    >
      ID
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="datum"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="110"
    >
      Datum ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="ophaal_locatie"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="260"
    >
      Ophaallocatie ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="bestemming"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="260"
    >
      Bestemming ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="chauffeur_toegewezen"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="140"
    >
      Chauffeur
      <br />
      toegewezen ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="naam_chauffeur"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="140"
    >
      Naam
      <br />
      chauffeur ↕
    </TableHeaderColumn>
    <TableHeaderColumn
      className=""
      columnClassName=""
      columnTitle={false}
      csvFieldType="string"
      dataAlign="left"
      dataField="naam_client"
      dataSort={true}
      defaultASC={false}
      editColumnClassName=""
      editable={true}
      expandable={true}
      filterFormatted={false}
      headerTitle={true}
      hidden={false}
      hiddenOnInsert={false}
      invalidEditColumnClassName=""
      isKey={false}
      keyValidator={false}
      searchable={true}
      sortIndicator={true}
      width="140"
    >
      Naam client ↕
    </TableHeaderColumn>
  </BootstrapTable>
  <CSVLink
    className="btn btn-primary crud-btn"
    data={
      Array [
        Object {
          "bestemming": "Herenlaan 23, 3701 AR, Zeist",
          "chauffeur_toegewezen": "Nee",
          "datum": "22-02-2017",
          "id": 1,
          "naam_chauffeur": "-",
          "naam_client": "Gerda Willems",
          "ophaal_locatie": "2e Hogeweg 46, 3709 AZ, Zeist",
        },
        Object {
          "bestemming": "Singelpark 7, 3984 NC, Odijk",
          "chauffeur_toegewezen": "Nee",
          "datum": "21-02-2017",
          "id": 2,
          "naam_chauffeur": "-",
          "naam_client": "Hans de Wit",
          "ophaal_locatie": "Wegedoorn 23, 3984 AZ, Odijk",
        },
        Object {
          "bestemming": "Herenlaan 23, 3701 AR, Zeist",
          "chauffeur_toegewezen": "Ja",
          "datum": "20-02-2017",
          "id": 3,
          "naam_chauffeur": "Taxi bedrijf",
          "naam_client": "Gerda Willems",
          "ophaal_locatie": "2e Hogeweg 46, 3709 AZ, Zeist",
        },
      ]
    }
    filename="zorginstelling_overview"
    separator=","
    target=""
    uFEFF={true}
  >
    Export als CSV
  </CSVLink>
</div>
`;
